engine/engine.go

```diff
@@ -416,7 +418,17 @@ func (r *ruleEngine) createViolation(conditionResponse ConditionResponse, rule R
 			incident.Message = templateString
 		}
 
-		incidents = append(incidents, incident)
+		// Checking duplication of incident
+
+		jsonIncident, _ := json.Marshal(incident)  // Converting to JSON to be used as keys of a map
+		stringJsonIncident := string(jsonIncident) // Converting JSON to string
+
+		_, isDuplicate := incidentsSet[stringJsonIncident] // Checking if incident already exists in set
```

Comment:
I usually perfer to make this a one line, but this is just a nit.

```
if _, ok := ...... ; !ok {
  do x
  }
  ```

Reply:
> I usually perfer to make this a one line, but this is just a nit.
> 
> ```
> if _, ok := ...... ; !ok {
>   do x
>   }
> ```

ah okay, will do that from next time. thank you!